Resources/Prototypes/Entities/Objects/Weapons/Guns/Pistols/pistols.yml

```diff
@@ -102,6 +102,41 @@
       gun_magazine: !type:ContainerSlot
       gun_chamber: !type:ContainerSlot
 
+- type: entity
+  name: echis
+  parent: BaseItem
+  id: WeaponPistolEchis
+  description: A viper for use by cyborgs. Creates .35 ammo on the fly from an internal ammo fabricator, which slowly self-charges.
+  components:
+  - type: Gun
+    fireRate: 5
+    selectedMode: SemiAuto
+    availableModes:
+      - SemiAuto
+      - FullAuto
+    soundGunshot:
+      path: /Audio/Weapons/Guns/Gunshots/pistol.ogg
+  - type: Sprite
+    sprite: Objects/Weapons/Guns/Pistols/viper.rsi
+    layers:
+      - state: base
+        map: ["enum.GunVisualLayers.Base"]
+      - state: mag-0
+        map: ["enum.GunVisualLayers.Mag"]
+  - type: ContainerContainer
+    containers:
+      ballistic-ammo: !type:Container
+  - type: ProjectileBatteryAmmoProvider
+    proto: CartridgePistolCaseless #So that borgs don't litter the ground with infinite bullet casings
+    fireCost: 100
+  - type: Battery
+    maxCharge: 1000
+    startingCharge: 1000
```

Comment:
Are you using cartridges because you actually want it to use cartridges or because you don't know how to make it shoot bullets directly?